spec/Section 3 -- Type System.md

```diff
@@ -553,11 +554,15 @@ of rules must be adhered to by every Object type in a GraphQL schema.
             the interface field type is either an Interface type or a Union type
             and the object field type is a possible type of the interface field
             type.
-         3. An object field type is a valid sub-type if it is a List type and
+         3. An object field type is a valid sub-type if it is an Interface type
+            and the interface field type is either an Interface type or a Union
+            type and the object field type is a possible type of the interface
+            field type.
```

Comment:
I think this language would need to be tweaked slightly:

> An object field type is a valid sub-type if it is an Interface type and the interface field type is either an Interface type or a Union type and _every possible type of the_ object field type is _also_ a possible type of the interface field type.

For example:
```graphql
interface Pet {
  mother: Pet
}

interface Canine {
  breed: String
}

type Wolf implements Canine {
  breed: String
}

type Labrador implements Canine, Pet {
  breed: String

  # Invalid: One of the possible types of Canine (Wolf) is not a possible type of Pet
  mother: Canine
}
```

Reply:
The weird thing is that this rule (which seems necessary) would cause a brittle base class problem: You might add a new type to the schema that implements some interfaces, and not others, and suddenly it causes _some other part of the schema_ to be invalid.

Reply:
Doesn't the rule apply to this scenario as written? `Canine` is not a possible type of `Pet`, so `mother: Canine` is invalid in `Labrador`.

Reply:
I worded it that way because I wasn't thinking of an Interface type as being a possible type for another Interface type. But based on the language for [fragment spreads](http://facebook.github.io/graphql/October2016/#GetPossibleTypes()):

> If _type_ is an interface type, return the set of types implementing _type_

Then I think I'd actually agree with the language as written. Under the definition I proposed, this would be valid, but it would not be valid with the language in the PR:

```graphql
interface Pet {
  mother: Pet
}

interface Canine {
  breed: String
}

# +Pet interface
type Wolf implements Canine, Pet {
  breed: String
  mother: Wolf
}

type Labrador implements Canine, Pet {
  breed: String

  # Valid under my definition: Every possible type of Canine is also a possible type of Pet
  # Invalid under original: Canine _itself_ is not a possible type of Pet
  mother: Canine
}
```

So... I retract my suggestion 😄  

I think the wording as-is, assuming that the definition of "possible types" is the defined by [`GetPossibleTypes`](http://facebook.github.io/graphql/October2016/#GetPossibleTypes()) function in section 5.4.2.3, is better than what I was proposing.